regparser/tree/xml_parser/tree_utils.py

```diff
@@ -110,33 +111,29 @@ def get_node_text(node, add_spaces=False):
     _superscript_to_text(node, add_spaces)
     _footnotes_to_text(node, add_spaces)
 
-    parts = [node.text] +\
-        list(chain(*([c.text, c.tail] for c in node.getchildren()))) +\
-        [node.tail]
+    parts = [node.text] + list(
+        chain(*([c.text, c.tail] for c in node.getchildren())))
 
     final_text = ''
     for part in filter(bool, parts):
         final_text = _combine_with_space(final_text, part, add_spaces)
     return final_text.strip()
 
 
-def get_node_text_tags_preserved(node):
-    """ Given an XML node, generate text from the node, skipping the PRTPAGE
-    tag. """
+_tag_black_list = ('PRTPAGE', )
 
-    html_parser = HTMLParser.HTMLParser()
 
-    if node.text:
-        node_text = node.text
-    else:
-        node_text = ''
+def get_node_text_tags_preserved(node):
+    """Get the body of an XML node as a string, avoiding a specific blacklist
+    of bad tags."""
+    node = deepcopy(node)
+    etree.strip_tags(node, *_tag_black_list)
+    node_text = etree.tostring(node)
 
-    for c in node:
-        if c.tag == 'E':
-            # xlmns non-sense makes me do this.
-            node_text += mtypes.emphasize(c.text)
-        if c.tail is not None:
-            node_text += c.tail
+    # Remove the wrapping tag
+    end_open_tag = node_text.index('>')
+    start_close_tag = node_text.rindex('<')
+    node_text = node_text[end_open_tag + 1:start_close_tag]
```

Comment:
Curious; do you think that way of removing the outer tag is better than this?
```
    parts = ([node.text] + list(itertools.chain(*([c.text, etree.tostring(c), c.tail] for c in node.getchildren()))) + [node.tail])
    node_text =  ''.join(filter(None, parts))```

(Which we use in line 113 above except without the final `.tail`.) The three lines are a little clearer, but also feel clunky and provoked a “there must be a cleaner way to do innerHTML()” response that led to a bunch of looking around on my part.